src/i8042.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Portions Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the THIRD-PARTY file.
+
+//! Provides emulation for a super minimal i8042 controller.
+//!
+//! This emulates just the CPU reset command.
+
+use std::fmt::{self, Display, Formatter};
+use std::{io, result};
+use vmm_sys_util::eventfd::EventFd;
+
+// Offset of the status port (port 0x64).
+const OFS_STATUS: u8 = 4;
```

Comment:
Maybe it's a bit clearer if we explicitly state in the comment that offset 4 represents the status register when reading, and the command register when writing (i.e. when sending the reset command).